packages/nimble-components/src/menu/tests/menu.spec.ts

```diff
@@ -1,11 +1,64 @@
+import { html } from '@microsoft/fast-element';
 import { Menu, menuTag } from '..';
+import { menuItemTag } from '../../menu-item';
+import { anchorMenuItemTag } from '../../anchor-menu-item';
+import { iconCheckTag } from '../../icons/check';
+import { Fixture, fixture } from '../../utilities/tests/fixture';
+import { waitForUpdatesAsync } from '../../testing/async-helpers';
 
 describe('Menu', () => {
+    async function setup(): Promise<Fixture<Menu>> {
+        return fixture<Menu>(html`
+            <${menuTag}>
+                <${menuItemTag}>Item 1</${menuItemTag}>
+                <${anchorMenuItemTag}>Item 2</${anchorMenuItemTag}>
+            </${menuTag}>
+        `);
+    }
+
     it('should export its tag', () => {
         expect(menuTag).toBe('nimble-menu');
     });
 
     it('can construct an element instance', () => {
         expect(document.createElement('nimble-menu')).toBeInstanceOf(Menu);
     });
+
+    it('properly indents items when an icon is slotted or removed after connecting', async () => {
```

Comment:
Should this test be split into multiple tests? It could be split by adding vs removing and for menu items vs anchor menu items. There are quite a few different things all be tested in this single test right now (though I understand it's all following the same code path).